opac/webapp/utils/utils.py

```diff
@@ -573,3 +585,72 @@ def fix_journal_last_issue(journal):
         **leg_dict).get_issue_seg()
     return journal.last_issue
 
+
+def render_citation(csl_json, style='apa', formatter=formatter.html, validate=False):
+    """
+    Given a csl_json and return a citation
+
+    Link do the csl-json schema: https://github.com/citation-style-language/schema/blob/master/schemas/input/csl-data.json
+
+    Example of csl_json param: 
+        [
+            {
+                "id": "wNZLxRjKfGdDw8KGmbNN7qj",
+                "DOI": "10.1590/0001-3765202020181115",
+                "URL": "http://dx.doi.org/10.1590/0001-3765202020181115",
+                "author": [
+                    {
+                        "family": "SANTOS-SILVA",
+                        "given": "JULIANA"
+                    },
+                    {
+                        "family": "ARAÚJO",
+                        "given": "TAINAR J."
+                    }
+                ],
+                "container-title": "Scientific Electronic Library Online",
+                "container-title-short": "SciELO",
+                "issue": "An. Acad. Bras. Ciênc., 2020 92(2)",
+                "issued": {
+                    "date-parts": [
+                        [
+                            2020,
+                            9
+                        ]
+                    ]
+                },
+                "page": "",
+                "publisher": "Academia Brasileira de Ciências",
+                "title": "Are Fabaceae the principal super-hosts of galls in Brazil?",
+                "title-short": "Are Fabaceae the principal super-hosts of galls in Brazil?",
+                "type": "research-article",
+                "volume": "92"
+            }
+        ]
+
+    Return a list citation as string.
+    
+    """
+
+    bib_source = CiteProcJSON(csl_json)
+
+    style_path = get_style_filepath(style)
+
+    bib_style = CitationStylesStyle(style_path, validate=validate)
+
+    bibliography = CitationStylesBibliography(bib_style, bib_source, formatter)
+
+    # Loop to the citations id on csl_json
+    ids = [c.get('id') for c in csl_json]
+
+    for id in ids:
```

Comment:
@gitnnolabs usar só o `for`. Mas não entendi por há o bloco `for`. Não é 1 como citar por artigo?

Reply:
@robertatakenaka pode ser mais de um item no csl_json. 

Exemplo:

```python
[{'id': 'wNZLxRjKfGdDw8KGmbNN7qj', 'DOI': '10.1590/0001-3765202020181115', 'URL': 'http://dx.doi.org/10.1590/0001-3765202020181115', 'author': [{'family': 'SANTOS-SILVA', 'given': 'JULIANA'}, {'family': 'ARAÚJO', 'given': 'TAINAR J.'}], 'container-title': 'Scientific Electronic Library Online', 'container-title-short': 'SciELO', 'issue': 'An. Acad. Bras. Ciênc., 2020 92(2)', 'issued': {'date-parts': [[2020, 9]]}, 'page': '', 'publisher': 'Academia Brasileira de Ciências', 'title': 'Are Fabaceae the principal super-hosts of galls in Brazil?', 'title-short': 'Are Fabaceae the principal super-hosts of galls in Brazil?', 'type': 'research-article', 'volume': '92'},{'id': 'wNZLxRjKfGdDw8KGmbNN7qr', 'DOI': 'asdasd', 'URL': 'sdfsfds', 'author': [{'family': 'SANTOS-SILVA', 'given': 'JULIANA'}, {'family': 'ARAÚJO', 'given': 'TAINAR J.'}], 'container-title': 'Scientific Electronic Library Online', 'container-title-short': 'SciELO', 'issue': 'An. Acad. Bras. Ciênc., 2020 92(2)', 'issued': {'date-parts': [[2020, 9]]}, 'page': '', 'publisher': 'Academia Brasileira de Ciências', 'title': 'Are Fabaceae the principal super-hosts of galls in Brazil?', 'title-short': 'Are Fabaceae the principal super-hosts of galls in Brazil?', 'type': 'research-article', 'volume': '92'}]
```